File: frontend/types/convites.ts
export type ConvitePublico = {
  id: string;
  familia_id: string;
  familia_nome: string;
  email: string;
  papel: "administrador" | "membro" | string;
  expira_em: string;
  convidado_por_nome: string;
};

export type ConviteAceito = {
  familia_id: string;
  mensagem: string;
};

File: backend/app/schemas/convites.py
from datetime import datetime

from pydantic import BaseModel, Field


class ConvitePublicoResponse(BaseModel):
    id: str
    familia_id: str
    familia_nome: str
    email: str
    papel: str
    expira_em: datetime
    convidado_por_nome: str


class AceitarConviteTokenRequest(BaseModel):
    token: str = Field(min_length=32, max_length=256)


class ConviteAceitoResponse(BaseModel):
    familia_id: str
    mensagem: str
